app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js

```diff
@@ -820,4 +822,51 @@ acceptance("Sidebar - Plugin API", function (needs) {
       resetCustomCategorySectionLinkPrefix();
     }
   });
+
+  test("Customizing the prefix icon used in a tag section link for a particular tag", async function (assert) {
+    try {
+      return await withPluginApi(PLUGIN_API_VERSION, async (api) => {
+        updateCurrentUser({
+          display_sidebar_tags: true,
+          sidebar_tags: [
+            { name: "tag2", pm_only: false },
+            { name: "tag1", pm_only: false },
+            { name: "tag3", pm_only: false },
+          ],
+        });
+
+        api.registerCustomTagSectionLinkPrefixIcon({
+          tagName: "tag1",
+          prefixValue: "wrench",
+          prefixColor: "#FF0000", // rgb(255, 0, 0)
+        });
+
+        await visit("/");
+
+        assert.ok(
+          exists(
+            `.sidebar-section-link[data-tag-name="tag1"] .prefix-icon.d-icon-wrench`
+          ),
+          "wrench icon is displayed for tag1 section link's prefix icon"
+        );
+
+        assert.strictEqual(
+          query(
+            `.sidebar-section-link[data-tag-name="tag1"] .sidebar-section-link-prefix`
+          ).style.color,
+          "rgb(255, 0, 0)",
+          "tag1 section link's prefix icon has the right color"
+        );
+
+        assert.ok(
+          exists(
+            `.sidebar-section-link[data-tag-name="tag2"] .prefix-icon.d-icon-tag`
+          ),
+          "default tag icon is displayed for tag2 section link's prefix icon"
+        );
+      });
+    } finally {
```

Comment:
Why not do this in `after` for the test rather than doing `try...finally`?

Reply:
Do you mean in an after hook? In this case, only this test needs to reset the global state so I’m calling it in a finally block.

Reply:
Thanks, Joffrey also explained this to me in person 😆 Just looked weird at first, all good 👍